### PR TITLE
Permalink

### PR DESCRIPTION
**Pull request by alan-wu**
Implementation to provide a way to get and set state of the app.

### CODE REVIEW

**Comment by alan-wu** [comment]
This is not ready for merging yet.

File: src/components/FloatingFlow.vue [hunk 1]
```diff
@@ -214,10 +221,31 @@ export default {
...
+      this.mainTabName = state.mainTabName;
+      this.zIndex = state.zIndex;
+      this.showDialogIcons = state.showDialogIcons;
+      this.dockedArray = [];
+      Object.assign(this.dockedArray, state.dockedArray);
+      this.activeDockedId = state.activeDockedId;
+      this.currentCount = state.currentCount;
+      this.entries = [];
+      Object.assign(this.entries, state.entries);
+      let closeItems = document.querySelectorAll('.mapboxgl-popup-close-button');
```
Comment (Tehsurfer) [inline src/components/FloatingFlow.vue:236]: Could probably switch this from `document` to a `$ref` right?

File: src/components/FloatingFlow.vue [hunk 2]
```diff
@@ -214,10 +221,31 @@ export default {
       this.bringDialogToFront(id);
     },
     resetApp: function(){
-      Object.assign(this.$data, initialState());
-      var closeItems = document.querySelectorAll('.mapboxgl-popup-close-button');
+      this.setState(initialState());
+    },
+    setState: function(state){
```
Comment (Tehsurfer) [inline src/components/FloatingFlow.vue:226]: Would it be cleaner to save all variables that are in a state in their own object? 

like as in:

```javascript
this.state = {
mainTabName: ...
z-index: ...
dockedArray: ...
}
```
Then we could do a single `Object.assign`
```javascript
Object.assign(this.state, state);
```

File: src/components/FloatingFlow.vue [hunk 3]
```diff
@@ -214,10 +221,31 @@ export default {
...
+      Object.assign(this.dockedArray, state.dockedArray);
+      this.activeDockedId = state.activeDockedId;
+      this.currentCount = state.currentCount;
+      this.entries = [];
+      Object.assign(this.entries, state.entries);
+      let closeItems = document.querySelectorAll('.mapboxgl-popup-close-button');
       closeItems.forEach( (item) => { item.click() });
     },
+    getState: function() {
+      let state = JSON.parse(JSON.stringify(this.$data));
```
Comment (Tehsurfer) [inline src/components/FloatingFlow.vue:240]: is the purpose of this to break the refence to `this.$data`?

Could use `let state = ...this.$data` if so.

**Review by Tehsurfer — APPROVED** [review]
Looks good to me! I added a few comments that I was thinking about for in future but nothing that would stop me from approving the changes.

File: src/components/FloatingFlow.vue (hunk 3 above)
Comment (alan-wu) [reply src/components/FloatingFlow.vue:240]: No, it is to copy without any of the vue reactive properties.

**Pull request merged**